src/Plugin/Condition/DataComparison.php

```diff
@@ -21,14 +21,14 @@
  *       label = @Translation("Data to compare"),
  *       description = @Translation("The data to be checked to be empty, specified by using a data selector, e.g. 'node:uid:entity:name:value'.")
  *     ),
+ *     "value" = @ContextDefinition("any",
+ *        label = @Translation("Data value"),
+ *        description = @Translation("The value to compare the data with.")
+ *     ),
  *     "operator" = @ContextDefinition("string",
  *       label = @Translation("Operator"),
  *       description = @Translation("The comparison operator."),
  *       required = FALSE
```

Comment:
This turns out be a very subtle bug.  Going through the debugger, I'm discovering that when RulesConditionBase::evaluate() calls $this->getContexts(), the ordering is NOT according to the order as defined in the annotation.  It's according to the order that the plugin is notified about the contexts at run time.  So you're not seeing annotation order; you're seeing the order in which $this->context was written to.  So order can get swapped.

If you want to make sure that you're getting things in annotation order, then you need to fill in the array you pass back really *is* the order given back by $this->getContextDefinitions().  The function needs to be rewritten as

drupal/core/lib/Drupal/Component/Plugin/ContextAwarePluginBase.php, line 79:
```php
  public function getContexts() {
    // Make sure all context objects are initialized.
    $ordered_contexts = [];
    foreach ($this->getContextDefinitions() as $name => $definition) {
      $ordered_contexts[$name] = $this->getContext($name);
    }
    return $ordered_contexts;
  }
```
This file is in Core, and needs to be reported against Core.  For our purposes, we should probably just override the function in our plugin base class.

But if I make this change, the old tests work absolutely fine.